### PR TITLE
DUOS-1389[risk=no]Remove unnecessary API call from Dar Application page

### PR DESCRIPTION
**Pull request by raejohanek**
## SCOPE:
- Remove getPropertiesByResearcherId call and instead use the properties on the already retrieved user from the getMe call

ADDRESSES:
https://broadworkbench.atlassian.net/browse/DUOS-1389

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/pages/DataAccessRequestApplication.js [hunk 1]
```diff
@@ -185,14 +185,6 @@ class DataAccessRequestApplication extends Component {
     return datasets;
   }
 
-  async getSOs()  {
```
Comment (raejohanek) [inline src/pages/DataAccessRequestApplication.js:188]: unused

**Review by JVThomas — DISMISSED** [review]
None of the researcher fields are populating in the DAR application
Screen Shot 2021-08-06 at 10 38 52 AM

Here's the current develop branch in comparison
Screen Shot 2021-08-06 at 10 40 55 AM

NOTE: eraCommonsId works fine, I just wasn't logged in when I took these screenshots.

File: src/libs/ajax.js [hunk 2]
```diff
@@ -800,6 +800,7 @@ export const PendingCases = {
 
 export const Researcher = {
 
+  //unused do we want to deprecate this API
```
Comment (rushtong) [inline src/libs/ajax.js]: Yes, we do need to deprecate that API.

File: src/pages/DataAccessRequestApplication.js [hunk 3]
```diff
@@ -213,45 +205,36 @@ class DataAccessRequestApplication extends Component {
       formData = Storage.getData('dar_application') === null ? this.state.formData : Storage.getData('dar_application');
       Storage.removeData('dar_application');
     }
-    let currentUserId = Storage.getCurrentUser().dacUserId;
-    let rpProperties = await Researcher.getPropertiesByResearcherId(currentUserId);
-    formData.darCode = fp.isNil(formData.darCode) ? null : formData.darCode;
-    formData.partialDarCode = fp.isNil(formData.partialDarCode) ? null : formData.partialDarCode;
+    let rpProperties = getPropertyValuesFromUser(researcher);
```
Comment (rushtong) [inline src/pages/DataAccessRequestApplication.js:208]: This method doesn't populate the completed property, so add that in.

**Review by rushtong — CHANGES_REQUESTED** [review]
See minor validation bug fix in comment

File: src/pages/DataAccessRequestApplication.js [hunk 4]
```diff
@@ -1028,7 +1011,7 @@ class DataAccessRequestApplication extends Component {
               div({ className: 'dialog-description' },
                 ['Are you sure you want to save this Data Access Request? Previous changes will be overwritten.'])
             ]),
-            div({ isRendered: this.state.step === 1 }, [
+            div({ isRendered: this.state.step === 1 && (this.state.formData.researcher !== '') }, [
```
Comment (rushtong) [inline src/pages/DataAccessRequestApplication.js]: From the console:
```
src/pages/DataAccessRequestApplication.js
  Line 1014:100:  Trailing spaces not allowed  no-trailing-spaces
```

File: src/pages/DataAccessRequestApplication.js [hunk 5]
```diff
@@ -213,45 +205,36 @@ class DataAccessRequestApplication extends Component {
...
+    formData.pubmedId = rpProperties.pubmedID;
+    formData.scientificUrl = rpProperties.scientificURL;
+    formData.userId = researcher.dacUserId;
 
     let completed = false;
     if (!fp.isNil(formData.darCode)) {
       completed = '';
-    } else if (rpProperties.completed !== undefined) {
-      completed = JSON.parse(rpProperties.completed);
+    } else if (rpProperties.completed !== '') {
```
Comment (rushtong) [inline src/pages/DataAccessRequestApplication.js:236]: `rpProperties.completed` can still be nil or undefined if `getPropertyValuesFromUser` wasn't able to find a matching property. In fact, I think that's true of all of the `rpProperties` fields, so all of the simplifications above should be reverted.

**Review by rushtong — CHANGES_REQUESTED** [review]
Works well, but we've lost a lot of nil checks that are still necessary.

File: src/pages/DataAccessRequestApplication.js (hunk 5 above)
Comment (raejohanek) [reply src/pages/DataAccessRequestApplication.js:236]: if the property is null then the helper returns the empty string.  utils.getPropertyValuesFromUser calls utils.findPropertyValue which does the null checking and replacing

Comment (rushtong) [reply src/pages/DataAccessRequestApplication.js:236]: Ah, thank you, yes, that works.

**Review by rushtong — APPROVED** [review]
